src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -74,14 +82,20 @@ public virtual string ApplicationInsightsInstrumentationKey
             set => UpdateSettingInCache(EnvironmentSettingNames.AppInsightsInstrumentationKey, value);
         }
 
+        public void SetConfigurationFactory(Func<IConfiguration> configurationRootFactory)
+        {
+            _configurationFactory = configurationRootFactory;
+            Reset();
+        }
+
         private string GetSettingFromCache(string settingKey)
         {
             if (string.IsNullOrEmpty(settingKey))
             {
                 throw new ArgumentNullException(nameof(settingKey));
             }
 
-            return _settingsCache.GetOrAdd(settingKey, (key) => Utility.GetSettingFromConfigOrEnvironment(key));
+            return _settingsCache.GetOrAdd(settingKey, (key) => GetSetting(key));
```

Comment:
Can you explain this change to setting resolution? We used to pull config OR environment

Reply:
Now we just pull from the underlying configuration object, which may be constructed differently depending on the host. We still register environment variables and config file providers, so that behavior is essentially the same, but other hosts (e.g. CLI) can register additional providers, supporting configuration coming from command line arguments, environment config files, etc.